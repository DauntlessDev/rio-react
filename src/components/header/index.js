import React, { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
    Background, Container, Feature, Text, Link as TextLink, ButtonLink, Logo, Darken, Inner, Group, FeatureCallOut, FeatureCallOutBottom, Dropdown,
    SearchIcon,
    SearchInput, Search, SubText,
    Picture, Profile, PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
    return bg ? (
        <Background {...restProps} data-testid="header-bg">
            <Darken>
                {children}
            </Darken>
        </Background>
    ) : <>{children}</>
}


Header.Profile = function HeaderProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>;
};


Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}
Header.FeatureCallOutBottom = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOutBottom {...restProps}>{children}</FeatureCallOutBottom>
}

Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
}

Header.SubText = function HeaderText({ children, ...restProps }) {
    return <SubText {...restProps}>{children}</SubText>
}


Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}


Header.Inner = function HeaderInner({ children, ...restProps }) {
    return <Inner {...restProps}>{children}</Inner>
}



Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>
}



Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <TextLink {...restProps}>{children}</TextLink>
}



Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReactRouterLink to={to}><Logo {...restProps}></Logo></ReactRouterLink>
    )
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [searchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps}>
            <SearchIcon onClick={() => setSearchActive((searchActive) => !searchActive)} data-testid="search-click">
                <img src="/images/icons/search.png" alt="Search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={searchActive}
                data-testid="search-input"
            />
        </Search>
    );
};


Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
    return <PlayButton {...restProps}>{children}</PlayButton>;
};

