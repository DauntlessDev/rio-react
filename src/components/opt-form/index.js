import React from 'react'
import { Container, Input, Button, Text } from './styles/opt-form'

export default function OptForm({ children, ...restProps }) {
    return (
        <Container{...restProps}>
            {children}
        </Container>
    )
}

OptForm.Input = function OutFormInput({ ...restProps }) {
    return <Input {...restProps} />
}

OptForm.Button = function OutFormButton({ children, ...restProps }) {
    return (
        <Button {...restProps}>
            {children}
            <img src="/images/icons/chevron-right.png" alt="Try Now" />
        </Button>
    )
}

OptForm.Text = function OutFormText({ children, ...restProps }) {
    return (
        <Text {...restProps}>
            {children}
        </Text>
    )
}



OptForm.Break = function OutFormBreak({ ...restProps }) {
    return (
        <Text {...restProps} />
    )
}

