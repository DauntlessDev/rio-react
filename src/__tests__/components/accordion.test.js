import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import faqsData from '../../fixtures/faqs';
import { Accordion } from '../../components';

describe('<Accordion />', () => {
    it('renders the <Accordion /> with populated data', () => {
        const { container, getByText } = render(
            <Accordion>
                <Accordion.Title>Frequently Asked Questions</Accordion.Title>
                <Accordion.Frame>
                    {faqsData.map((item) => (
                        <Accordion.Item key={item.id}>
                            <Accordion.Header>{item.header}</Accordion.Header>
                            <Accordion.Body>{item.body}</Accordion.Body>
                        </Accordion.Item>
                    ))}
                </Accordion.Frame>
            </Accordion>
        );

        expect(getByText('Frequently Asked Questions')).toBeTruthy();
        expect(getByText('What is Rioflix?')).toBeTruthy();
        expect(getByText('How much does Rioflix cost?')).toBeTruthy();
        expect(getByText('Where can I watch?')).toBeTruthy();
        expect(getByText('How do I cancel?')).toBeTruthy();
        expect(getByText('What can I watch on Rioflix?')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('opens and closes the <Accordion /> component', () => {
        const { container, queryByText } = render(
            <Accordion>
                <Accordion.Title>Frequently Asked Questions</Accordion.Title>
                <Accordion.Frame>
                    {faqsData.map((item) => (

                        <Accordion.Item key={item.id}>
                            <Accordion.Header>{item.header}</Accordion.Header>
                            <Accordion.Body data-testid="accordion-body">{item.body}</Accordion.Body>
                        </Accordion.Item>
                    ))}
                </Accordion.Frame>
            </Accordion>
        );

        const bodyText = "Watch Rioflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from Â£5.99 a month. No extra costs or contracts."

        expect(queryByText(bodyText)).toBeFalsy();
        fireEvent.click(queryByText("How much does Rioflix cost?"));
        expect(queryByText(bodyText)).toBeTruthy();

        fireEvent.click(queryByText("How much does Rioflix cost?"));
        expect(queryByText(bodyText)).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });
});
