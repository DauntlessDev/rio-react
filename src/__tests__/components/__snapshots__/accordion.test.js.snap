// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-9zbxzv-0 xLYsw"
>
  <div
    class="accordion__Inner-sc-9zbxzv-2 iyakfd"
  >
    <h1
      class="accordion__Title-sc-9zbxzv-3 dccKDK"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-sc-9zbxzv-1 hvKCqN"
    >
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          What is Rioflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          How much does Rioflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          What can I watch on Rioflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-9zbxzv-0 xLYsw"
>
  <div
    class="accordion__Inner-sc-9zbxzv-2 iyakfd"
  >
    <h1
      class="accordion__Title-sc-9zbxzv-3 dccKDK"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-sc-9zbxzv-1 hvKCqN"
    >
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          What is Rioflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          How much does Rioflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-9zbxzv-4 jOCFeu"
      >
        <div
          class="accordion__Header-sc-9zbxzv-5 ejacmQ"
        >
          What can I watch on Rioflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
